File: lib_common/src/main/java/com/wujia/lib_common/base/baseadapter/CommonAdapter.java
package com.wujia.lib_common.base.baseadapter;

import android.content.Context;
import android.view.LayoutInflater;

import com.wujia.lib_common.base.baseadapter.base.ItemViewDelegate;
import com.wujia.lib_common.base.baseadapter.base.ViewHolder;

import java.util.List;

public abstract class CommonAdapter<T> extends MultiItemTypeAdapter<T> {
    protected Context mContext;
    protected int mLayoutId;
    protected LayoutInflater mInflater;

    public CommonAdapter(final Context context, final int layoutId, List<T> datas) {
        super(context, datas);
        mContext = context;
        mInflater = LayoutInflater.from(context);
        mLayoutId = layoutId;
        mDatas = datas;

        addItemViewDelegate(new ItemViewDelegate<T>() {
            @Override
            public int getItemViewLayoutId() {
                return layoutId;
            }

            @Override
            public boolean isForViewType(T item, int position) {
                return true;
            }

            @Override
            public void convert(ViewHolder holder, T t, int position) {
                CommonAdapter.this.convert(holder, t, position);
            }
        });
    }

    protected abstract void convert(ViewHolder holder, T t, int position);

}

File: lib_common/src/main/java/com/wujia/lib_common/base/baseadapter/CommonAdapter.kt
package com.wujia.lib_common.base.baseadapter

import android.content.Context
import android.view.LayoutInflater

import com.wujia.lib_common.base.baseadapter.base.ItemViewDelegate
import com.wujia.lib_common.base.baseadapter.base.ViewHolder

abstract class CommonAdapter<T>(  mContext: Context, protected var mLayoutId: Int, datas: List<T>) : MultiItemTypeAdapter<T>(mContext, datas) {
    private var mInflater: LayoutInflater = LayoutInflater.from(mContext)

    init {
        this.datas = datas

        addItemViewDelegate(object : ItemViewDelegate<T> {
            override val itemViewLayoutId: Int
                get() = mLayoutId

            override fun isForViewType(item: T, position: Int): Boolean {
                return true
            }

            override fun convert(holder: ViewHolder, t: T, position: Int) {
                this@CommonAdapter.convert(holder, t, position)
            }
        })
    }

    protected abstract fun convert(holder: ViewHolder, t: T, position: Int)

}
